flink-table/flink-table-common/src/test/java/org/apache/flink/table/types/inference/InputTypeStrategiesTest.java

```diff
@@ -154,19 +154,41 @@
 				.expectSignature("f(STRING)\nf(INT)\nf(BOOLEAN)")
 				.expectArgumentTypes(DataTypes.INT().bridgedTo(int.class)),
 
-			// OR with implicit casting
 			TestSpec
 				.forStrategy(
+					"OR with implicit casting",
 					or(
 						explicitSequence(DataTypes.TINYINT()),
 						explicitSequence(DataTypes.INT()),
 						explicitSequence(DataTypes.BIGINT())))
 				.calledWithArgumentTypes(DataTypes.SMALLINT())
 				.expectArgumentTypes(DataTypes.INT()),
 
-			// invalid type in OR
 			TestSpec
-				.forStrategy(or(explicitSequence(DataTypes.INT()), explicitSequence(DataTypes.STRING())))
+				.forStrategy(
```

Comment:
Shall we add a test that check the first input strategy is chosen if multiple match with an implicit cast?